### PR TITLE
Optional TaskContext added to DruidBeamMaker

### PR DESCRIPTION
**Pull request by dgolitsyn**

### CODE REVIEW

**Review by drcrallen — COMMENTED** [review]
Unit tests?

File: core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala [hunk 1]
```diff
@@ -130,6 +132,10 @@ class DruidBeamMaker[A](
         log.warn(s"DruidTuning key[$k] for task[$taskId] overridden from[$v] to[${druidTuningMapWithOverrides(k)}].")
       }
     }
+    val taskContextMap = taskContext match {
+      case taskContext if taskContext.nonEmpty => Map("context" -> taskContext)
+      case other => Map.empty
```
Comment (logarithm) [inline core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala]: if you are not interested in the case value you can replace it by `_`.

File: core/src/main/scala/com/metamx/tranquility/druid/DruidBeams.scala [hunk 2]
```diff
@@ -714,7 +719,8 @@ object DruidBeams
     _beamMergeFn: Option[Seq[Beam[EventType]] => Beam[EventType]] = None,
     _alertMap: Option[Dict] = None,
     _objectWriter: Option[ObjectWriter[EventType]] = None,
-    _timestamper: Option[Timestamper[EventType]] = None
+    _timestamper: Option[Timestamper[EventType]] = None,
+    _taskContext: Option[Map[String, Any]] = None
```
Comment (logarithm) [inline core/src/main/scala/com/metamx/tranquility/druid/DruidBeams.scala]: `Map[String, Any]` might be replaced by `Dict`

File: core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala [hunk 3]
```diff
@@ -48,15 +48,17 @@ class DruidBeamMaker[A](
   indexService: IndexService,
   emitter: ServiceEmitter,
   objectWriter: ObjectWriter[A],
-  druidObjectMapper: ObjectMapper
+  druidObjectMapper: ObjectMapper,
+  taskContext: Map[String, Any] = Map.empty
```
Comment (logarithm) [inline core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala]: `Map[String, Any]` might be a `Dict` (as we already use it for `druidTuningMap`)

Comment (logarithm) [reply core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala]: Why do we need default value for `taskContext` when in `DruidBeams` we already have default value for `taskContext` variable?

**Review by logarithm — CHANGES_REQUESTED** [review]
Is `taskContext` intended to be used for other purposes except `label` ?

**Review by egor-ryashin — APPROVED** [review]

File: core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala [hunk 4]
```diff
@@ -130,6 +132,10 @@ class DruidBeamMaker[A](
         log.warn(s"DruidTuning key[$k] for task[$taskId] overridden from[$v] to[${druidTuningMapWithOverrides(k)}].")
       }
     }
+    val taskContextMap = taskContext match {
+      case taskContext if taskContext.nonEmpty => Map("context" -> taskContext)
```
Comment (logarithm) [inline core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala]: Using the same name in pattern matching as original value (on which we do pattern matching) is suspicious and in some cases might lead to errors.

File: core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala (hunk 3 above)
Comment (dgolitsyn) [reply core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala]: fixed

File: core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala (hunk 1 above)
Comment (dgolitsyn) [reply core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala]: fixed

File: core/src/main/scala/com/metamx/tranquility/druid/DruidBeams.scala (hunk 2 above)
Comment (dgolitsyn) [reply core/src/main/scala/com/metamx/tranquility/druid/DruidBeams.scala]: fixed

**Review by logarithm — APPROVED** [review]

**Pull request merged**